src/f4enix/input/MCNPinput.py

```diff
@@ -229,6 +261,50 @@ def __init__(
         # cells.extend(surfs)
         # cells.extend(data)
 
+    @property
+    def cells(self) -> dict[str, parser.Card]:
+        """Get the cells attribute."""
+        return self._cells
+
+    @cells.setter
+    def cells(self, value: dict[str, parser.Card]) -> None:
+        """Set the cells attribute."""
+        # enforce that the keys are all corresponding to the cell number
+        # in case of difference, the dictionary key takes precedence
+        for key, card in value.items():
+            if key != str(card.name):
+                logging.warning(
+                    f"The cell number {card.name} is different from Key {key} and will be replaced"
+                )
+                card.name = int(key)
```

Comment:
Here I would not rely on cell.name, but on the first value of values attribute of the cell Card. The reason is because a typical scenario is that you apply a renumbering with our renumber method and then add cells. The renumber doesn't modify cell.name, but only the value in values. this may block adding new cells. Solution: let renumber modify cell.name or don't use cell.name at all

Reply:
Well I think this proves the effectiveness of the current implementation.
From what you are saying, it may happen that the 'cel' value has not been changed but not the other way around. When I find a descrepancy in the name, I impose both the card.name and the 'cel' (or 'sur') value to be equal to the key so whe should be all good